scripts/setup-ubuntu.sh

```diff
@@ -0,0 +1,263 @@
+#PKGS="$PKGS libavcodec-dev libavformat-dev libavutil-dev"
+PKGS="$PKGS libncurses5-dev libncursesw5-dev"
+PKGS="$PKGS libsqlite3-dev libssl-dev"
+
+# Add yarn before NodeSource so it can all go in one update
+#
+yv=$(getVersion yarn "NA")
+if [ ${yv} = "NA" ]; then
+	curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg |\
+			 sudo apt-key add -
+	echo "deb https://dl.yarnpkg.com/debian/ stable main" |\
+			 sudo tee /etc/apt/sources.list.d/yarn.list
+	PKGS="$PKGS yarn"
+fi
+
+# Check for minimum node version
+#
+nv=$(getVersion node 0)
+if ! versionAtLeast "$nv" "$NODE_MIN_VERSION"; then
+	# Nodesource will also make sure we have apt-transport-https
+	# and will run apt-get-update when done
+	#
+	curl -sL https://deb.nodesource.com/setup_${NODE_MIN_VERSION}.x |
+								sudo -E bash -
+	PKGS="$PKGS nodejs"
+	DO_UPDATE=0
+else
+	PKGS="$PKGS apt-transport-https"
+fi
+
+# Run apt-get update if still necessary
+#
+if [ ${DO_UPDATE:-1} ]; then sudo apt-get update -y; fi
+
+# Install necessary pakackes
+#
+sudo apt-get install -y $PKGS
+sudo apt-get autoremove -y
+
+# Check for ffmpeg and install by building if necessary
+# This needs to be done before building moonfire so it can
+# find the right versions of the libraries.
+#
+ffv=`ffmpeg -version 2>/dev/null | extractVersion libavutil`
+ffv=${ffv:-0}
+if ! versionAtLeast "$ffv" "$FFMPEG_MIN_VERSION"; then
+	if [ "${DONT_BUILD_FFMPEG:-0}" -ne 0 ]; then
+		echo "ffmpeg version (${ffv}) installed is too old for moonfire."
+		echo "Suggest you manually install at least version $FFMPEG_MIN_VERSION of libavutil."
+		echo "ffmpeg versions 2.x and 3.x all should work."
+	else
+		OLDDIR=`pwd`
+		cd ..
+		if [ -d FFmpeg ]; then
+			echo "Removing older FFmpeg directory..."; echo
+			rm -fr FFmpeg
+		fi
+		echo "Fetching FFmpeg source..."; echo
+		git clone --depth 1 -b "release/${FFMPEG_RELEASE_VERSION}" https://github.com/FFmpeg/FFmpeg.git
+		cd FFmpeg
+		pt=`uname -p 2>& /dev/null`
+		if [ -z "${pt##*86*}" ]; then
+			sudo apt-get install -y yasm
+		fi
+		./configure --enable-shared
+		make
+		sudo make install
+		sudo ldconfig
+		cd "$OLDDIR"
+		OLDDIR=
+	fi
+else
+	echo "FFmpeg is already usable..."; echo
+fi
+
+# If cargo appears installed, initialize for using it so rustc can be found
+#
+initCargo
+
+# Make sure we have rust and cargo
+rv=$(getVersion rustc 0.0)
+if ! versionAtLeast "$rv" "$RUSTC_MIN_VERSION"; then
+	echo "Installing latest rust and cargo..."; echo
+	curl https://sh.rustup.rs -sSf | sh -s - -y
+	initCargo
+fi
+
+cv=$(getVersion cargo "NA")
+if [ ${cv} = "NA" ]; then
+	echo "Cargo is not (properly) installed, but rust is."
+	echo "Suggest you install the latest rustup, or manually install cargo."
+	echo "Install using: curl https://sh.rustup.rs -sSf | sh -s -y"
+	exit 1
+fi
+
+# Now make sure we have dev environment and tools for the UI portion
+#
+echo "Adding yarn components we need..."; echo
+grep -qv \"webpack\" package.json && yarn add --dev webpack@${WEBPACK_MIN_VERSION}
```

Comment:
This manipulates the checked-in package.json and yarn.lock. I'd rather we do that directly than have people run scripts that do so. Let's remove this and the matching lines in script-functions.sh.